massa-execution-worker/src/execution.rs

```diff
@@ -1341,4 +1356,54 @@ impl ExecutionState {
     pub fn get_address_future_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
         context_guard!(self).get_address_future_deferred_credits(address, self.config.thread_count)
     }
+
+    /// Initialize the hashmap of addresses from the vesting file
+    fn init_vesting_registry(
+        config: &ExecutionConfig,
+    ) -> Result<PreHashMap<Address, Vec<VestingRange>>, ExecutionError> {
+        let mut hashmap: PreHashMap<Address, Vec<VestingRange>> = serde_json::from_str(
```

Comment:
Instead of having mut here and mut in the `for` why not using a map on your initialization to have no mutability

Reply:
if we want to modify end_slot on a VestingRange we need mut on map() too